wrphttp/requestResponse.go

```diff
@@ -105,16 +139,38 @@ type entityResponseWriter struct {
 	f wrp.Format
 }
 
-func (erw *entityResponseWriter) WriteWRP(v interface{}) (int, error) {
-	var (
-		output  []byte
-		encoder = wrp.NewEncoderBytes(&output, erw.f)
-	)
+func (erw *entityResponseWriter) WriteWRP(e *Entity) (int, error) {
+	if e == nil {
+		return 0, ErrUndefinedEntity
+	}
+
+	var output []byte
 
-	if err := encoder.Encode(v); err != nil {
-		return 0, err
+	if len(e.Bytes) > 0 && e.Format == erw.f {
+		output = e.Bytes
+	} else {
+		encoder := wrp.NewEncoderBytes(&output, erw.f)
+
+		if err := encoder.Encode(e.Message); err != nil {
+			return 0, err
+		}
 	}
 
 	erw.ResponseWriter.Header().Set("Content-Type", erw.f.ContentType())
 	return erw.ResponseWriter.Write(output)
 }
+
+func (erw *entityResponseWriter) WriteWRPFromBytes(f wrp.Format, encodedWRP []byte) (int, error) {
```

Comment:
As a general naming convention, words like "from" are avoided in `golang` APIs.  This can be just `WriteWRPBytes`.